src/components/molecules/Slider/style.scss

```diff
@@ -1,34 +1,56 @@
 .slider {
-    position: relative;
     width: 100%;
-    height: 140vw;
+    height: auto;
+    overflow: hidden;
+    box-sizing: border-box;
+    display: flex;
+    align-items: center;
 
-    &-img {
-        position: absolute;
-        transition: left 0.5s;
-        left: 0%;
+    &__slide {
+        position: relative;
+        min-width: 100%;
+        height: auto;
+        overflow: hidden;
     }
 
-    &-img-prev {
-        position: absolute;
-        left: -100%;
-        transition: left 0.5s;
-        &-trans {
-            position: absolute;
-            left: 100%;
-            transition: left 0.5s;
-          }
+    &__slide0-transitionOn {
+        @extend .slider__slide;
+        transform: translateX(0%);
+        transition: 0.3s;
     }
 
-    &-img-next {
-        position: absolute;
-        left: 100%;
-        transition: left 0.5s;
-        &-trans {
-          position: absolute;
-          left: 100%;
-          transition: left 0.5s;
-        }
+    &__slide-100-transitionOn {
+        @extend .slider__slide;
+        transform: translateX(-100%);
+        transition: 0.3s;
+    }
+
+    &__slide-100-transitionOff {
+        @extend .slider__slide;
+        transform: translateX(-100%);
+    }
+
+    &__slide-200-transitionOn {
+        @extend .slider__slide;
+        transform: translateX(-200%);
+        transition: 0.3s;
+    }
+
+    &__slide-300-transitionOn {
+        @extend .slider__slide;
+        transform: translateX(-300%);
+        transition: 0.3s;
+    }
+
+    &__slide-300-transitionOff {
+        @extend .slider__slide;
+        transform: translateX(-300%);
+    }
+
+    &__slide-400-transitionOn {
+        @extend .slider__slide;
+        transform: translateX(-400%);
+        transition: 0.3s;
     }
```

Comment:
If you were to have 10 slides in the carousel will you need to add more transitions or those 4 will be enough? This approach with multiple classes(__slide-400, __slide-300 ...) is not the way to go if you will have to add new class for every additional carousel element

Reply:
If i use classNames, i need to add new class for new slide, in this case.
But if I don't want to add a new class every time, I could use inline styles and pass transition and transform values as props. This is possible option.